gph/src/ripser.h

```diff
@@ -706,11 +706,13 @@ class ripser
                                         index_t n, OutputIterator out) const
     {
         --n;
-        for (index_t k = dim + 1; k > 0; --k) {
+        for (index_t k = dim + 1; k > 1; --k) {
             n = get_max_vertex(idx, k, n);
             *out++ = n;
             idx -= binomial_coeff(n, k);
         }
+        // k = 1 is 0-simplices (vertices), and get_max_vertex(idx, 1, n) = idx
+        *out = idx;
```

Comment:
This has also a non-trivial impact. Compared to original implementation, when adding the last element, you still move the iterator by one.
Here we don't do it anymore, so I also think this add a certain amount of speed-up.

Reply:
I'd be curious to see the performance difference without `++` and with it!